core/src/services/hdfs/backend.rs

```diff
@@ -223,10 +224,9 @@ fn tmp_file_of(path: &str) -> String {
 /// Backend for hdfs services.
 #[derive(Debug, Clone)]
 pub struct HdfsBackend {
-    pub info: Arc<AccessorInfo>,
-    pub root: String,
+    root: String,
```

Comment:
Move this into `HdfsCore`. When using the root in the `HdfsBackend`, use `self.core.root`.